dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -1,501 +1,491 @@
-            if (categoryNotified != null) {
-                for (List<URL> us : categoryNotified.values()) {
-                    for (URL u : us) {
-                        if (buf.length() > 0) {
-                            buf.append(URL_SEPARATOR);
-                        }
-                        buf.append(u.toFullString());
-                    }
-                }
-            }
-            properties.setProperty(url.getServiceKey(), buf.toString());
-            long version = lastCacheChanged.incrementAndGet();
-            if (syncSaveFile) {
-                doSaveProperties(version);
-            } else {
-                registryCacheExecutor.execute(new SaveProperties(version));
-            }
-        } catch (Throwable t) {
-            logger.warn(t.getMessage(), t);
-        }
-    }
-
-    @Override
-    public void destroy() {
-        if (logger.isInfoEnabled()) {
-            logger.info("Destroy registry:" + getUrl());
-        }
-        Set<URL> destroyRegistered = new HashSet<URL>(getRegistered());
-        if (!destroyRegistered.isEmpty()) {
-            for (URL url : new HashSet<URL>(getRegistered())) {
-                if (url.getParameter(Constants.DYNAMIC_KEY, true)) {
-                    try {
-                        unregister(url);
-                        if (logger.isInfoEnabled()) {
-                            logger.info("Destroy unregister url " + url);
-                        }
-                    } catch (Throwable t) {
-                        logger.warn("Failed to unregister url " + url + " to registry " + getUrl() + " on destroy, cause: " + t.getMessage(), t);
-                    }
-                }
-            }
-        }
-        Map<URL, Set<NotifyListener>> destroySubscribed = new HashMap<URL, Set<NotifyListener>>(getSubscribed());
-        if (!destroySubscribed.isEmpty()) {
-            for (Map.Entry<URL, Set<NotifyListener>> entry : destroySubscribed.entrySet()) {
-                URL url = entry.getKey();
-                for (NotifyListener listener : entry.getValue()) {
-                    try {
-                        unsubscribe(url, listener);
-                        if (logger.isInfoEnabled()) {
-                            logger.info("Destroy unsubscribe url " + url);
-                        }
-                    } catch (Throwable t) {
-                        logger.warn("Failed to unsubscribe url " + url + " to registry " + getUrl() + " on destroy, cause: " + t.getMessage(), t);
-                    }
-                }
-            }
-        }
-    }
-
-    @Override
-    public String toString() {
-        return getUrl().toString();
-    }
-
-    private class SaveProperties implements Runnable {
-        private long version;
-
-        private SaveProperties(long version) {
-            this.version = version;
-        }
-
-        @Override
-        public void run() {
-            doSaveProperties(version);
-        }
-    }
-
-}
+/*
```

Comment:
Why are you changing the entire file?

Reply:
It's likely the behaviours of the IDE, I'll update to avoid this

Reply:
@khanimteyaz That's wired, I check it in the GitHub diff page and there is no such problem, are you sure you are looking at the last commit.

Reply:
@kezhenxu94 Please check https://github.com/apache/incubator-dubbo/pull/3102/files , this issue still exist. All the lines of AbstractRegistry has been changed.

Reply:
@ralf0131 Thanks, I've updated again, now I saw `+98 −125` and it should be fixed now.